client/store/product.js

```diff
@@ -0,0 +1,47 @@
+import axios from 'axios'
+
+//ACTION TYPES
+const FETCH_PRODUCTS = 'FETCH_PRODUCTS'
+const FETCH_SINGLE_PRODUCT = 'FETCH_SINGLE_PRODUCT'
+
+//ACTION CREATORS
+const fetchedProducts = products => ({type: FETCH_PRODUCTS, products})
+const fetchedSingleProduct = product => ({type: FETCH_SINGLE_PRODUCT, product})
+
+//THUNK CREATORS
+export const fetchProductsThunk = () => async dispatch => {
+  try {
+    const {data} = await axios.get(`api/products`)
+    dispatch(fetchedProducts(data))
+  } catch (error) {
+    console.error(error)
+  }
+}
+
+export const fetchSingleProductThunk = id => async dispatch => {
+  try {
+    const {data} = await axios.get(`api/products/${id}`)
+    dispatch(fetchedSingleProduct(data))
+  } catch (error) {
+    console.error(error)
+  }
+}
+
+//INITIAL STATE
+const initialState = {
+  products: []
+}
+
+//REDUCER
+const productReducers = (state = initialState, action) => {
+  switch (action.type) {
+    case FETCH_PRODUCTS:
+      return action.products
+    case FETCH_SINGLE_PRODUCT:
+      return action.product
+    default:
+      return state
+  }
+}
+
```

Comment:
Currently the reducer will replace the initialState object with an array for the FETCH_PRODUCTS case and an object with just on product from the FETCH_SINGLE_PRODUCTS case.  This would change the way we would access the data. For example to get products on the initialState it would be initialState.products, but after this reducer runs the initialState would only be the array or only the single product retrieved from the db.  I think it would be better to change the return statements to something like: return {...state, products} for the FETCH_PRODUCTS case.  The FETCH_SINGLE_PRODUCT case can either replace the products array with an array that just has the one product or add a new key  for the single product.